bokeh/models/annotations.py

```diff
@@ -432,6 +431,74 @@ class Span(Annotation):
     The %s values for the span.
     """)
 
+class Title(Annotation):
+    """ Render a text box as an annotation.
+
+    """
+
+    x = Float(help="""
+    The x-coordinate in screen coordinates to locate the text anchors.
+    """)
+
+    y = Float(help="""
+    The y-coordinate in screen coordinates to locate the text anchors.
+    """)
+
+    text = String(help="""
+    The text value to render.
+    """)
+
+    angle = Angle(default=0, help="""
+    The angle to rotate the text, as measured from the horizontal.
+
+    .. warning::
+        The center of rotation for canvas and css render_modes is different.
+        For `render_mode="canvas"` the label is rotated from the top-left
+        corner of the annotation, while for `render_mode="css"` the annotation
+        is rotated around it's center.
+    """)
+
+    angle_units = Enum(AngleUnits, default='rad', help="""
+    Acceptable values for units are ``"rad"`` and ``"deg"``
+    """)
+
+    text_props = Include(TextProps, use_prefix=False, help="""
+    The %s values for the text.
+    """)
+
+    text_font_size = Override(default={"value": "14pt"})
+
+    text_font_style = Override(default="bold")
+
+    background_props = Include(FillProps, use_prefix=True, help="""
+    The %s values for the text bounding box.
+    """)
+
+    background_fill_color = Override(default=None)
+
+    border_props = Include(LineProps, use_prefix=True, help="""
+    The %s values for the text bounding box.
+    """)
+
+    border_line_color = Override(default=None)
+
+    render_mode = Enum(RenderMode, default="canvas", help="""
```

Comment:
Should render mode be available? Probably. One of the things layout is going to need to do is to measure the extent of the title so it can adjust the panel size accordingly - is that going to be equally easy on canvas and css mode - probably - just thinking out loud.